### DISCUSSION TITLE
More than one admin user?

### DISCUSSION THREAD

**Discussion by jackgit28** [benjaminjonard/koillection#706] [Ideas]
Hello!

Firstly, I really like this project. I've deployed this to my k3s self-hosted environment and found it to be just the thing my wife was looking for to manage her book collection. Simple and clean.

After setting her up as the admin, and then making myself another different user out of curiosity, I realised I couldn't find a way to recover/reset the admin account password should she ever lose/forget the password for it. I also couldn't promote my user to an admin via the UI using her user. So I had a look directly at the database (postgres in my case) to find another way to potentially recover a lost admin account.

**TL;DR** - I promoted my non admin user to an admin by editing it's _roles_ value in the _koi_user_ table to match that of the existing admin user. It worked, but is there any technical reason why this would be a bad idea? (If all admins trust each other)

Thanks,
Jack.

**Comment by benjaminjonard** [comment]
Hello, 

It should be perfectly fine to have several administrator users, there no technical limitations related to that.
I will add the possibility to set a user as administrator if it can be useful.  

But you are right, there is no real way to recover the password if the administrator forgets it. I think the best option might be to add a new shell command that could set a new password to any user based on their username.

**Comment by jackgit28** [comment]
Hi Benjamin,

Thank you for your reply. Either of those two ideas would be useful and appreciated.
